### ISSUE TITLE
Address stacks and envs by path

### ISSUE THREAD

**Issue by jamesroutley** [cloudreach/sceptre#62]
## Implementation

We currently build stacks and environments with the syntax:

```
sceptre <action>-stack env stack
sceptre <action>-env env
```

This proposal would change the addressing to:

```
sceptre <action> path/to/stack.yaml
sceptre <action> --recursive path/to/env/
```

## Pros
- By supplying file paths, we get native tab completion (only tested on `zsh` on `macOS`)

## Cons

**Comment by theseanything** [comment]
Pro this! But not sure about the `--recursive` flag.

**Comment by mitom** [comment]
Instead of `--recursive` could we have just:
```
sceptre <action> path/to/env/
```

As in passing in a directory would be the same as `launch-env` and passing in a file would be the same as `launch-stack`?

**Comment by theseanything** [comment]
Yeah that would be ideal. But we need to put some sort of protection around doing actions to a whole environment. It would be too easy for a user to accidentally create or update stacks unintentionally. CloudFormation slip-ups can be costly :(

**Comment by jamesroutley** [comment]
@seanrankine what about it are you not sure about?

**Comment by theseanything** [comment]
its in an awkward position and the word `recursive` is long... Just means its not super easy to switch between a stack or env command in the terminal.

**Comment by jamesroutley** [comment]
oh yeah I think we'd do a `-r` as well.

**Comment by theseanything** [comment]
Yup but still not a huge fan of it sitting in between the command and path. It'd be nice to just tack it on the end of a command

**Comment by mitom** [comment]
I'd rather have it in front, `-r` is not too bad and flags generally come in front of positional arguments, not after.

Also, to avoid potentially harmful/accidental actions you could add a simple verification 
`You are about to update the entire <env> envrionment, do you want to continue? [Y|n]`. It would make it easy to switch between commands (no other arguments need to change) and it would make it clear what is going to happen. For CI tools a `-y` flag could assume yes.

**Comment by theseanything** [comment]
I like the verification step way of doing it and the safety. Thoughts @jamesroutley? Also would we want to put .yaml at the end of a stack? I'd prefer to omit it.

**Comment by jamesroutley** [comment]
I've opened an issue about adding verification to the delete commands #61. Should we add one for other commands as well? Maybe for "dangerous" commands like `update`?

I put the `.yaml` for a couple of reasons:

1. Sceptre fundamentally deals with files. Each stack maps onto a single file. I think it's beneficial to emphasise this by getting users to supply a path to a file.
2. Using an actual file path gives you tab-completion. This completes to `/path/stack.yaml`, so if we didn't have the `.yaml` users could tab in the filename but would have to delete those five chars
3. It makes the `yaml` vs `yml` thing ( #67  ) a bit more explicit and removes some ambiguity about which one we use.
